### PR TITLE
Support configuration of external links at job level.

### PR DESCRIPTION
**Pull request by ypadron-in**
External links are an optional config in azkaban.properties file. They are rendered as buttons in the UI.
We're adding the ability to configure external links at job level in addition to flow level.
Here is how they are defined and would look like:
Screenshot 2023-04-18 at 10 40 20 PM

```
azkaban.server.job.external.analyzer.topics=jobExternalLink1, jobExternalLink2
azkaban.server.job.external.analyzer.jobExternalLink1.label=Job External Link 1
azkaban.server.job.external.analyzer.jobExternalLink1.url=https://job.external.link1.com/params?azkabanFlowUrl=${url}

azkaban.server.job.external.analyzer.jobExternalLink2.url=https://job.external.link1.com/params?exec=${exec_id}&ws=${webserver_host}&job=${job_id}
azkaban.server.job.external.analyzer.jobExternalLink2.label=Job External Link 2
```
Screenshot 2023-04-18 at 10 41 22 PM

```
azkaban.server.external.analyzer.topics=externalLink1,externalLink2

azkaban.server.external.analyzer.externalLink1.url=https://external-link1.com/params?execution_id=${exec_id}
azkaban.server.external.analyzer.externalLink1.label=My External Link

azkaban.server.external.analyzer.externalLink2.url=https://external-link2.com/params?az_url=${url}
azkaban.server.external.analyzer.externalLink2.label=Test
```

### CODE REVIEW

File: az-core/src/main/java/azkaban/Constants.java [hunk 1]
```diff
@@ -264,35 +265,25 @@ public static class ConfigurationKeys {
...
 
-    // External URL template of a given topic, specified in the list defined above
-    //Deprecated, it is replaced by AZKABAN_SERVER_EXTERNAL_ANALYZER_TOPIC_URL
-    public static final String AZKABAN_SERVER_EXTERNAL_TOPIC_URL = "azkaban.server.external.${topic}.url";
-
-    // Designates one of the external link topics to correspond to an execution analyzer
-    //Deprecated, replaced by AZKABAN_SERVER_EXTERNAL_ANALYZER_TOPICS
-    public static final String AZKABAN_SERVER_EXTERNAL_ANALYZER_TOPIC = "azkaban.server.external.analyzer.topic";
-    //Deprecated, it is replaced by AZKABAN_SERVER_EXTERNAL_ANALYZER_TOPIC_LABEL
-    public static final String AZKABAN_SERVER_EXTERNAL_ANALYZER_LABEL = "azkaban.server.external.analyzer.label";
```
Comment (ypadron-in) [inline az-core/src/main/java/azkaban/Constants.java:275]: Deleting long deprecated config keys

File: az-core/src/main/java/azkaban/utils/Props.java [hunk 2]
```diff
@@ -481,17 +482,11 @@ public List<String> getStringListFromCluster(final String key) {
...
+    if (val == null || (val = val.trim()).length() == 0) {
       return Collections.emptyList();
     }
-
-    if (containsKey(key)) {
-      return Arrays.asList(val.split(sep));
-    } else {
-      throw new UndefinedPropertyException("Missing required property '" + key
-          + "'");
-    }
```
Comment (ypadron-in) [inline az-core/src/main/java/azkaban/utils/Props.java:494]: if condition should be guaranteed to be true by lines 484 and 485 so deleting it.

File: azkaban-common/src/main/java/azkaban/utils/ExternalLink.java [hunk 3]
```diff
@@ -46,4 +46,23 @@ public boolean isLinkUrlValid() {
...
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    ExternalLink link = (ExternalLink) o;
+    return Objects.equals(this.topic, link.topic) &&
+        Objects.equals(this.label, link.label) &&
+        Objects.equals(this.linkUrl, link.linkUrl) &&
```
Comment (byjiang1996) [inline azkaban-common/src/main/java/azkaban/utils/ExternalLink.java:60]: nit: It looks a little bit odd to use `Objects.equals` to test if two strings are equal. Personally prefer this.topic.equals(link.topic)

**Review by byjiang1996 — APPROVED** [review]

File: azkaban-common/src/main/java/azkaban/utils/ExternalLink.java (hunk 3 above)
Comment (ypadron-in) [reply azkaban-common/src/main/java/azkaban/utils/ExternalLink.java:60]: I was trying to avoid the case of `null.equals("someString")`

**Pull request merged**